src/index.js

```diff
@@ -132,19 +131,46 @@ function addTask() {
 // remove task
 
 // edit task
+function markTaskAsUncompleted(taskElement) {
+  taskElement.classList.remove('completed');
+  const checkbox = taskElement.querySelector('input[type="checkbox"]');
+  checkbox.checked = false;
+}
 
 function removeCompletedTasks() {
   const checkboxes = document.querySelectorAll('input[type="checkbox"]');
   checkboxes.forEach((checkbox) => {
-    if (checkbox.checked) {
+    checkbox.addEventListener('change', function () { // add event listener to checkbox
+      if (this.checked) {
+        const li = this.parentNode;
+        if (li.classList.contains('completed')) {
+          markTaskAsUncompleted(li);
+        }
+        removeTask(li);
+      }
+    });
+    if (checkbox.checked) { // check if checkbox is initially checked
       const li = checkbox.parentNode;
+      if (li.classList.contains('completed')) {
+        markTaskAsUncompleted(li);
+      }
       removeTask(li);
     }
   });
 }
 
 completeButton.addEventListener('click', removeCompletedTasks);
 
+function completeTasks() {
+  const tasks = Array.from(document.querySelectorAll('li'));
+  const completedTasks = tasks.filter((task) => task.classList.contains('completed'));
+  completedTasks.forEach((task) => {
+    removeTask(task);
+  });
+}
```

Comment:
- [ ] I have noticed that when the first completed tasks are cleared, clicking on the `checkbox` deletes the tasks instead of marking the task as completed. Kindly try to fix this so that the application is User-friendly.
- [ ] [OPTIONAL] Please ensure that the item's index is updated. This will help you to easily target the tasks for edition, and deletion.